x11-libs/fltk/fltk-1.3.8.ebuild

```diff
@@ -0,0 +1,200 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools flag-o-matic xdg-utils multilib-minimal
+
+DESCRIPTION="C++ user interface toolkit for X and OpenGL"
+HOMEPAGE="https://www.fltk.org/"
+SRC_URI="https://www.fltk.org/pub/${PN}/${PV}/${P}-source.tar.gz"
+
+SLOT="1"
+LICENSE="FLTK LGPL-2"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux"
+IUSE="cairo debug doc examples games +opengl static-libs +threads +xft +xinerama"
+
+RDEPEND="
+	media-libs/libjpeg-turbo:0=[${MULTILIB_USEDEP}]
+	>=media-libs/libpng-1.2:0=[${MULTILIB_USEDEP}]
+	sys-libs/zlib[${MULTILIB_USEDEP}]
+	x11-libs/libICE[${MULTILIB_USEDEP}]
+	x11-libs/libSM[${MULTILIB_USEDEP}]
+	x11-libs/libXcursor[${MULTILIB_USEDEP}]
+	x11-libs/libXext[${MULTILIB_USEDEP}]
+	x11-libs/libXfixes[${MULTILIB_USEDEP}]
+	x11-libs/libXt[${MULTILIB_USEDEP}]
+	cairo? ( x11-libs/cairo[${MULTILIB_USEDEP},X] )
+	opengl? (
+		virtual/glu[${MULTILIB_USEDEP}]
+		virtual/opengl[${MULTILIB_USEDEP}]
+	)
+	xft? ( x11-libs/libXft[${MULTILIB_USEDEP}] )
+	xinerama? ( x11-libs/libXinerama[${MULTILIB_USEDEP}] )
+"
+DEPEND="
+	${RDEPEND}
+	virtual/pkgconfig
+	x11-base/xorg-proto
+	doc? ( app-doc/doxygen )
+"
+DOCS=(
+	ANNOUNCEMENT
+	CHANGES
+	CHANGES_1.0
+	CHANGES_1.1
+	CREDITS
+	README
+	README.abi-version.txt
+	README.Cairo.txt
+	README.CMake.txt
+	README.MSWindows.txt
+	README.OSX.txt
+	README.Unix.txt
+)
+FLTK_GAMES="
+	blocks
+	checkers
+	sudoku
+"
+PATCHES=(
+	"${FILESDIR}"/${PN}-1.3.0-share.patch
+	"${FILESDIR}"/${PN}-1.3.4-conf-tests.patch
+	"${FILESDIR}"/${PN}-1.3.3-makefile-dirs.patch
+	"${FILESDIR}"/${PN}-1.3.3-visibility.patch
+)
+
+pkg_setup() {
+	unset FLTK_LIBDIRS
+}
+
+src_prepare() {
+	default
+
+	rm -rf zlib jpeg png || die
+
+	sed -i \
+		-e 's:@HLINKS@::g' FL/Makefile.in || die
+	sed -i \
+		-e '/x-fluid/d' fluid/Makefile || die
+	sed -i \
+		-e '/C\(XX\)\?FLAGS=/s:@C\(XX\)\?FLAGS@::' \
+		-e '/^LDFLAGS=/d' \
+		"${S}/fltk-config.in" || die
+	# docs in proper docdir
+	sed -i \
+		-e "/^docdir/s:fltk:${PF}/html:" \
+		-e "/SILENT:/d" \
+		makeinclude.in || die
+
+	# also in Makefile:config.guess config.sub:
+	cp misc/config.{guess,sub} . || die
+
+	cp fluid/{,x-}fluid.desktop || die
+
+	eaclocal
+	eautoconf
+	multilib_copy_sources
+}
+
+multilib_src_configure() {
+	unset FLTK_LIBDIRS
+
+	local FLTK_INCDIR=${EPREFIX}/usr/include/fltk
+	local FLTK_LIBDIR=${EPREFIX}/usr/$(get_libdir)/fltk
+	FLTK_LIBDIRS+=${FLTK_LIBDIRS+:}${FLTK_LIBDIR}
+
+	multilib_is_native_abi && use prefix &&
+		append-ldflags -Wl,-rpath -Wl,"${FLTK_LIBDIR}"
+
+	econf \
+		$(use_enable cairo) \
+		$(use_enable debug) \
+		$(use_enable opengl gl) \
+		$(use_enable threads) \
+		$(use_enable xft) \
+		$(use_enable xinerama) \
+		--disable-localjpeg \
+		--disable-localpng \
+		--disable-localzlib \
+		--docdir="${EPREFIX}/usr/share/doc/${PF}/html" \
+		--enable-largefile \
+		--enable-shared \
```

Comment:
This makes the `static-libs` use flag seem a bit redundant.

Reply:
Heh, of course it is :) 

I'll add some comment to the ebuild, thanks for explaining!

Reply:
Although I don't see a dependency for static-libs on either octave or tigervnc. If nothing **needs** the static libs enabled on this package, maybe we can unconditionally always just remove them?

Reply:
Sorry, I just misunderstood.

I didn't find any package that really depends on this ``static-lib`` useflag in ::gentoo. Only executables in ``test/``(from ``x11-libs/fltk`` test stage) are linking statically against libfltk.a. Therefore, we need to consider how to safely remove this  ``static-lib``  useflag. I may need to change the link method in ``test/``, or consider using ``test`` useflag to control it.